### ISSUE TITLE
redirect to white screen and not the app

### ISSUE THREAD

**Issue by iit2008047** [msintuneappsdk/intune-app-wrapping-tool-ios#68]
**Describe the bug:**
When I am launching the app for the first time, after loggin in and setting FACE ID .. I am redirect to white screen and not the app 

**To Reproduce**
https://user-images.githubusercontent.com/76396/142637892-a658759e-d564-4224-875b-af521f94ded9.mp4

**Expected behavior:**
The wrapped app should open

pen
**Screenshots and logs:**
[IntuneMAMDiagnosticFiles (1).txt](https://github.com/msintuneappsdk/intune-app-wrapping-tool-ios/files/7570821/IntuneMAMDiagnosticFiles.1.txt)

**Smartphone (please complete the following information):**
iOS devices 10.0 +

**Intune app wrapping tool (please complete the following information):**
Latest version 

**Additional context:**

**Comment by iit2008047** [comment]
@gastaffo i have opened a new issue

**Comment by gastaffo** [comment]
@iit2008047 So I can close the other issue?

**Comment by iit2008047** [comment]
Yes .. Please help me figure this issue out.

**Comment by iit2008047** [comment]
@gastaffo is there any update ??

**Comment by gastaffo** [comment]
Sorry no. A few things.
1. Does your repro require credentials other than Intune credentials? If not then you could send us an unwrapped IPA and we could wrap it in debug mode and attempt to see whats going on.
2. Could you attempt a repro with Verbose logging and then provide us the logs? To do this before wrapping the app add in the info.plist add a dictionary with the name "IntuneMAMSettings" inside that dictionary add a boolean called "VerboseLoggingEnabled" to YES.

**Comment by gastaffo** [comment]
@iit2008047  Ping for notification

**Comment by iit2008047** [comment]
@gastaffo can u share your email address .. will share

**Comment by gastaffo** [comment]
@iit2008047 gastaffo@microsoft.com

**Comment by iit2008047** [comment]
@gastaffo shared unwrapped IPA  via email

**Comment by iit2008047** [comment]
[IntuneMAMDiagnosticFiles (7).txt](https://github.com/msintuneappsdk/intune-app-wrapping-tool-ios/files/7631791/IntuneMAMDiagnosticFiles.7.txt)
@gastaffo this is the log collected after adding "IntuneMAMSettings" -> VerboseLoggingEnabled in plist

**Comment by gastaffo** [comment]
@iit2008047 Does the user have an MAM-CA policies assigned to them? Also is authenticator installed on the device?

**Comment by iit2008047** [comment]
<img width="788" alt="Screenshot 2021-12-02 at 9 24 48 AM" src="https://user-images.githubusercontent.com/76396/144354914-5c7eb4cc-6391-4736-a959-69f4bf904fd4.png">
<img width="718" alt="Screenshot 2021-12-02 at 9 24 58 AM" src="https://user-images.githubusercontent.com/76396/144354926-503872f8-a6bc-452f-8e42-e83d7ac05752.png">
<img width="737" alt="Screenshot 2021-12-02 at 9 25 06 AM" src="https://user-images.githubusercontent.com/76396/144354928-0306c4d6-7276-402a-be0e-3194aeea74fd.png">
<img width="975" alt="Screenshot 2021-12-02 at 9 25 12 AM" src="https://user-images.githubusercontent.com/76396/144354929-84dec8fd-41b5-4c3e-a039-fafe20f94997.png">

Here is the App protection policies.

**Comment by iit2008047** [comment]
authenticator is also installed in iOS device @gastaffo

**Comment by iit2008047** [comment]
Hey @gastaffo, Hope you are doing good .. please let me know know if you have any update?

**Comment by gastaffo** [comment]
@iit2008047 Trying to run with a debug wrapper

**Comment by gastaffo** [comment]
@iit2008047 After the SDK calls into the apps delegate for `applicationDidFinishLaunchWithOptions:` it stalls. Looking at the call stack we see this
  * frame #0: 0x00000001b8c90504 libsystem_kernel.dylib`mach_msg_trap + 8
    frame #1: 0x00000001b8c90b9c libsystem_kernel.dylib`mach_msg + 76
    frame #2: 0x000000018176b738 CoreFoundation`__CFRunLoopServiceMachPort + 372
    frame #3: 0x000000018176fa2c CoreFoundation`__CFRunLoopRun + 1212
    frame #4: 0x0000000181783468 CoreFoundation`CFRunLoopRunSpecific + 600
    frame #5: 0x0000000182f9fc94 Foundation`-[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 236
    frame #6: 0x0000000182fe0dc8 Foundation`-[NSRunLoop(NSRunLoop) runUntilDate:] + 92
    frame #7: 0x00000001044c8398 advocacyApp`___lldb_unnamed_symbol15832$$advocacyApp + 144
    frame #8: 0x0000000104263510 advocacyApp`___lldb_unnamed_symbol51$$advocacyApp + 756

which leaves us to believe the app is running its own run loop. We are guessing this run loop is waiting on some state to change that never does. Do you know what the loop is waiting for?

**Comment by iit2008047** [comment]
@gastaffo thanks for the update. 
we investigated further with your inputs.

The app is a react native app and we have added a custom splash screen handling till the RN bridge is initialized. 
After removing that splash screen handling, the white screen issue did not happen and the app started up as expected. 

Would u please help us understand how wrapper tool used applicationDidFinishLaunchWithOptions. Thanks

**Comment by gastaffo** [comment]
@iit2008047 The SDK will intercept the OS calling into applicationDidFinishLaunchingWithOptions for the Apps Delegate so we can apply conditional launch policy. Once we are done we then call into delegate. For apps built with things like React this can sometimes cause problems.

In the IntuneMAMSettings dictionary in the info.plist could you add "FinishLaunchingAtStartup" set to YES and retry with the splash screen? That changes the timing of the call and that might help React.

**Comment by gastaffo** [comment]
@iit2008047 Any updates? Is the wrapper working as expected now?

**Comment by iit2008047** [comment]
Thanks @gastaffo this was helpful ..

**Comment by gastaffo** [comment]
@iit2008047 No problem! I will close the opened issues. Also if you do not want to put "FinishingLaunchingAtStartup" in the info.plist directly you can also use the command line option `-finishLaunching` when wrapping.
